content/docs/pipelines/preemptible.md

```diff
@@ -0,0 +1,314 @@
+follow the guide to [deploying Kubeflow on GCP](/docs/gke/deploy/).
+
+## Using preemptible VMs with Kubeflow Pipelines
+
+In summary, the steps to schedule a pipeline to run on [preemptible
+VMs](https://cloud.google.com/compute/docs/instances/preemptible) are as
+follows: 
+
+1.  Create a
+    [node pool](https://cloud.google.com/kubernetes-engine/docs/concepts/node-pools)
+    in your cluster that contains preemptible VMs.
+1.  Configure your pipelines to run on the preemptible VMs.
+
+The following sections contain more detail about the above steps.
+
+### 1. Create a node pool with preemptible VMs
+
+Use the `gcloud` command to
+[create a node pool](https://cloud.google.com/sdk/gcloud/reference/container/node-pools/create).
+The following example includes placeholders to illustrate the important
+configurations:
+
+    gcloud container node-pools create PREEMPTIBLE_CPU_POOL \
+    	--cluster=CLUSTER_NAME \
+        --enable-autoscaling --max-nodes=MAX_NODES --min-nodes=MIN_NODES \
+        --preemptible \
+        --node-taints=preemptible=true:NoSchedule \
+        --service-account=DEPLOYMENT_NAME-vm@PROJECT_NAME.iam.gserviceaccount.com
+
+Where:
+
++   `PREEMPTIBLE_CPU_POOL` is the name of the node pool. 
++   `CLUSTER_NAME` is the name of the GKE cluster.
++   `MAX_NODES` and `MIN_NODES` are the maximum and minimum number of nodes
+    for the [GKE 
+    autoscaling](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-autoscaler)
+    functionality.
++   `DEPLOYMENT_NAME` is the name of your Kubeflow deployment. If you used
+    [the CLI to deploy Kubeflow](https://www.kubeflow.org/docs/gke/deploy/deploy-cli/),
+    this name is the value of the `${KFAPP}` environment variable. If you used
+    the [deployment UI](https://www.kubeflow.org/docs/gke/deploy/deploy-ui/),
+    this name is the value you specified as the deployment name.
++   `PROJECT_NAME` is the name of your GCP project.
+
+Below is an example of the command:
+
+    gcloud container node-pools create preemptible-cpu-pool \
+    	--cluster=user-4-18 \
+        --enable-autoscaling --max-nodes=4 --min-nodes=0 \
+        --preemptible \
+        --node-taints=preemptible=true:NoSchedule \
+        --service-account=user-4-18-vm@ml-pipeline-project.iam.gserviceaccount.com
+
+### 2. Schedule your pipeline to run on the preemptible VMs
+
+After configuring a node pool with preemptible VMs, you must configure your
+pipelines to run on the preemptible VMs. 
+
+In the [DSL code](https://www.kubeflow.org/docs/pipelines/sdk/dsl-overview/) for
+your pipeline, add the following to the `ContainerOp` instance:
+
+    .apply(gcp.use_preemptible_nodepool())
+
+The above function works for both methods of generating the `ContainerOp`:
+
++   The `ContainerOp` generated from 
+[`kfp.components.func_to_container_op`](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/components/_python_op.py).
```

Comment:
Github can generate a link pointing to the specific line, in this example:
https://github.com/kubeflow/pipelines/blob/c3235d725eb1d1eb06b5600a8291967aa6cf518f/sdk/python/kfp/components/_python_op.py#L283

Here is how to get the link above.
When you click the line, there will be a 'three dot' on the left; Then click it, and choose 'Copy Permalink'. 
The other benefit for the generated link is its version fixed.

Reply:
If someone adds or removes code to the file, the line number will change. Also, I'm not sure we want to fix on a specific version in the docs, because we'll need to update the links when we move to the next version.

I'm thinking I'll leave the link relatively vague - what do you think?

Reply:
If someone adds/removes code to the file, the line will not change in this version, though.
However, you have a good point. It does require some maintenance efforts in this case.
I'm OK with it being vague.
Thanks

Reply:
Thanks!